src/CLRunner.ts

```diff
@@ -335,6 +335,110 @@ export class CLRunner {
         return startResponse
     }
 
+    private async StartSession(appId: string, createParams: CLM.SessionCreateParams): Promise<CLM.Session> {
+
+        // Don't save logs on server if custom storage was provided
+        if (ConversationLearner.logStorage) {
+            createParams.saveToLog = false
+        }
+
+        const session = await this.clClient.StartSession(appId, createParams as CLM.SessionCreateParams)
+
+        // If using customer storage add to log storage
+        if (ConversationLearner.logStorage) {
+            // For self-hosted log storage logDialogId is sessionId
+            session.logDialogId = session.sessionId
+            const logDialog: CLM.LogDialog = {
+                logDialogId: session.logDialogId,
+                packageId: session.packageId,
+                rounds: [],
+                initialFilledEntities: [],
+                targetTrainDialogIds: [],
+                createdDateTime: new Date().toJSON(),
+                // Start out the same.  End is updated when dialog is edited
+                dialogBeginDatetime: new Date().toJSON(),
+                dialogEndDatetime: new Date().toJSON(),
+                lastModifiedDateTime: new Date().toJSON(),
+                metrics: ""
+            }
+            await ConversationLearner.logStorage.Add(appId, logDialog)
```

Comment:
>await ConversationLearner.logStorage.Add(appId, logDialog) [](start = 12, length = 58)

probably we shouldn't log the dialog if model level flag says not to log dialogs.

Reply:
Reasonable.